File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <bits/stdc++.h>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    stack<char> st;
    for(int i=0; i < s.length(); i++)
    {
        if(s[i] == '(')
            st.push('(');
        else if(s[i] == ')' && !st.empty())
            st.pop();
        else
        {
            answer = false;
            break;
        }
    }
    if(!st.empty())
        answer = false;
    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    st = []
    for i in s:
        if i == '(':
            st.append(1)
        elif st and i == ')':
            st.pop()
        else:
            answer = False
            break
    
    if st:
        answer = False
    return answer